apps/server/src/modules/task/uc/task.uc.spec.ts

```diff
@@ -142,998 +82,529 @@ describe('TaskService', () => {
-			const user = helper.getFirstUser();
-
-			const tasks = [];
-			const parents = [];
-
-			const mockRestore = findAllCompletedForStudentMocks(parents, tasks);
-			const spy = setTaskRepoMock.findAllCurrentByIds([]);
-
-			const skip = 0;
-			const limit = 10;
-
-			const expectedOptions: IFindOptions<Task> = {
-				pagination: { skip: 0, limit: 10 },
-				order: { dueDate: SortOrder.asc },
-			};
-
-			await service.findAllCompletedForStudent(user.id, { skip, limit });
-
-			expect(spy).toHaveBeenCalledWith([], [], expectedOptions);
+			expect(result.length).toEqual(1);
+			expect(result[0].status).toEqual({
+				graded: 2,
+				submitted: 2,
+				maxSubmissions: course.getNumberOfStudents(),
+			});
 
-			mockRestore();
-			spy.mockRestore();
+			spyTaskRepoFindAll.mockRestore();
+			spyGetPermittedCourses.mockRestore();
 		});
 
-		it('should compute status for tasks', async () => {
-			const helper = new TaskTestHelper();
-			const user = helper.getFirstUser();
-			const parent1 = helper.createTaskParent();
-			const task1 = helper.createTask(parent1.id);
-			const submission1 = helper.createSubmission(task1);
-
-			const tasks = [task1];
-			const parents = [parent1];
-			const submissions = [submission1];
+		it('should count only unique student ids of graded submissions', async () => {
+			const student1 = new UserTaskInfo({ firstName: 'John', lastName: 'Doe' });
+			student1.id = currentUser.userId;
+			const student2 = new UserTaskInfo({ firstName: 'Marla', lastName: 'Mathe' });
+			student2.id = new ObjectId().toHexString();
+			const course = new Course({ name: 'course #1', schoolId: new ObjectId() });
+			const task = new Task({ name: 'task #1', private: false, parent: course });
+			const submission1 = new Submission({ task, student: student1, comment: 'submission #1' });
+			const submission2 = new Submission({ task, student: student2, comment: 'submission #2' });
+			const submission3 = new Submission({ task, student: student2, comment: 'submission #3' });
+			task.submissions = new Collection<Submission>(task, [submission1, submission2, submission3]);
 
-			const mockRestore = findAllCompletedForStudentMocks(parents, tasks, submissions);
+			jest.spyOn(submission1, 'isGraded').mockImplementation(() => true);
+			jest.spyOn(submission2, 'isGraded').mockImplementation(() => true);
+			jest.spyOn(submission3, 'isGraded').mockImplementation(() => true);
+			const spyTaskRepoFindAll = setTaskRepoMock.findAll([task]);
+			const spyGetPermittedCourses = setAuthorizationServiceMock.getPermittedCourses();
 
 			const paginationQuery = new PaginationQuery();
-			const [result] = await service.findAllCompletedForStudent(user.id, paginationQuery);
+			const [result] = await service.findAll(currentUser, paginationQuery);
 
+			expect(result.length).toEqual(1);
 			expect(result[0].status).toEqual({
-				graded: 0,
-				submitted: 1,
-				maxSubmissions: 1,
+				graded: 2,
+				submitted: 2,
+				maxSubmissions: course.getNumberOfStudents(),
```

Comment:
Please do forget the student should only see maxSubmission 1.

Reply:
but that test is for a user with the teacher role right? the tests for the student role check for `maxSubmissions: 1`, e.g.
https://github.com/hpi-schul-cloud/schulcloud-server/blob/e70753720e208fb4df7e876af432c75ee94e3aca/apps/server/src/modules/task/uc/task.uc.spec.ts#L278